### PR TITLE
[4.0][com_privacy] - convert to prepared statement

### PR DESCRIPTION
**Pull request by alikon**
### Summary of Changes 
use prepared statement for SQL

### Testing Instructions
test com_privacy

### Expected result
should work as before

### Actual result
N/A

### CODE REVIEW

File: administrator/components/com_privacy/Model/ExportModel.php [hunk 1]
```diff
@@ -82,9 +83,10 @@ public function collectDataForExportRequest($id = null)
 
 		$userId = (int) $db->setQuery(
 			$db->getQuery(true)
-				->select('id')
+				->select($db->quoteName('id'))
 				->from($db->quoteName('#__users'))
-				->where($db->quoteName('email') . ' = ' . $db->quote($table->email)),
+				->where($db->quoteName('email') . ' = :email')
+				->bind(':email' . $table->email),
```
Comment (Quy) [inline administrator/components/com_privacy/Model/ExportModel.php]:
```suggestion
				->bind(':email', $table->email),
```

File: administrator/components/com_privacy/Model/ConsentsModel.php [hunk 2]
```diff
@@ -76,29 +77,37 @@ protected function getListQuery()
...
+				$search = '%' . substr($search, 5) . '%';
+				$query->where($db->quoteName('u.name') . ' LIKE :search')
+					->bind(':search', $search);
 			}
 			else
 			{
-				$search = $db->quote('%' . $db->escape($search, true) . '%');
-				$query->where('(' . $db->quoteName('u.username') . ' LIKE ' . $search . ')');
+				$search = '%' . $search . '%';
+				$query->where('(' . $db->quoteName('u.username') . ' LIKE :search )')
```
Comment (Quy) [inline administrator/components/com_privacy/Model/ConsentsModel.php]:
```suggestion
				$query->where('(' . $db->quoteName('u.username') . ' LIKE :search)')
```

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on b47e1276761b3804370042a17ad8b8125b5f27de<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26017">issues.joomla.org/tracker/joomla-cms/26017</a>.</sub>

File: components/com_privacy/Model/RequestModel.php [hunk 3]
```diff
@@ -11,6 +11,7 @@
 
 defined('_JEXEC') or die;
 
+use Exception;
```
Comment (mbabker) [inline components/com_privacy/Model/RequestModel.php]: Somebody make up their mind on a code style rule for global classes and enforce it with the tools.  There is no reason for some people to be using `\Exception` and some people to be importing it with `use`.

Comment (HLeithner) [reply components/com_privacy/Model/RequestModel.php]: Doesn't we import all classes?

Comment (alikon) [reply components/com_privacy/Model/RequestModel.php]: So what is the code style standard ?... i'll correct if i was wrong (better phpstorm done that work for me :smile: )

Comment (alikon) [reply components/com_privacy/Model/RequestModel.php]: hope cs ok now

File: components/com_privacy/Model/RequestModel.php [hunk 4]
```diff
@@ -247,8 +249,8 @@ public function getForm($data = [], $loadData = true)
 	 *
 	 * @return  Table  A JTable object
 	 *
+	 * @throws  Exception
```
Comment (Quy) [inline components/com_privacy/Model/RequestModel.php]:
```suggestion
	 * @throws  \Exception
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on e97cbf854bb840b7c7b08b4a9cf97658e0b94859<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26017">issues.joomla.org/tracker/joomla-cms/26017</a>.</sub>

File: administrator/components/com_privacy/Model/RequestModel.php [hunk 5]
```diff
@@ -436,10 +437,12 @@ public function validate($form, $data, $group = null)
...
 		$query = $db->getQuery(true)
 			->select('COUNT(id)')
-			->from('#__privacy_requests')
-			->where('email = ' . $db->quote($validatedData['email']))
-			->where('request_type = ' . $db->quote($validatedData['request_type']))
-			->where('status IN (0, 1)');
+			->from($db->quoteName('#__privacy_requests'))
+			->where($db->quoteName('email') . ' = :email')
+			->where($db->quoteName('request_type') . ' = :requesttype')
+			->where($db->quoteName('status'), [0, 1])
```
Comment (wilsonge) [inline administrator/components/com_privacy/Model/RequestModel.php]: This was supposed to be a `whereIn`

**Comment by wilsonge** [comment]
Conflicts too please

**Comment by anibalsanchez** [comment]
I have tested this item :white_check_mark: successfully on e97cbf854bb840b7c7b08b4a9cf97658e0b94859<br /><br />Test OK<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26017">issues.joomla.org/tracker/joomla-cms/26017</a>.</sub>

File: administrator/components/com_privacy/Model/RequestModel.php (hunk 5 above)
Comment (alikon) [reply administrator/components/com_privacy/Model/RequestModel.php]: fixed

**Comment by wilsonge** [comment]
Thanks!

**Comment by richard67** [comment]
@wilsonge This has PHPCS errors.

**Comment by richard67** [comment]
@wilsonge It is reported as PHPCS but it is not CS only ... I've pinged you on Glip, please check.

**Comment by richard67** [comment]
PR is #27041 . Please test.

**Pull request merged**